internal/controller/utils.go

```diff
@@ -45,6 +45,22 @@ func setDeploymentReplicas(deployment *appsv1.Deployment, replicas int32) bool {
 	return false
 }
 
+func setTolerations(deployment *appsv1.Deployment, tolerations []corev1.Toleration) bool {
+	if !apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.Tolerations, tolerations) {
```

Comment:
should the ! (negation) be there?

Reply:
Yes , the object is different than what is present , we update.